File: services/intelligence-api/app/schemas/scenario.py
"""Scenario schemas — the entry point for all intelligence flows."""
from enum import Enum
from pydantic import BaseModel, Field
from typing import Optional


class ScenarioSeverity(str, Enum):
    CRITICAL = "critical"
    HIGH = "high"
    MEDIUM = "medium"
    LOW = "low"


class ScenarioCategory(str, Enum):
    TRADE = "trade"
    ENERGY = "energy"
    GEOPOLITICAL = "geopolitical"
    FINANCIAL = "financial"
    REGULATORY = "regulatory"
    DEMAND = "demand"
    INFRASTRUCTURE = "infrastructure"
    CONFIDENCE = "confidence"


class GCCCountryCode(str, Enum):
    SA = "SA"
    AE = "AE"
    KW = "KW"
    QA = "QA"
    BH = "BH"
    OM = "OM"


class SectorCode(str, Enum):
    """Legacy sector codes — kept for backward compatibility.
    The canonical 4-tier SectorCode lives in schemas/sector.py.
    """
    # Tier 1 — Critical Sovereign Systems
    AVIATION = "AVIATION"
    OIL_GAS = "OIL_GAS"
    BANKING = "BANKING"
    ENERGY_INFRASTRUCTURE = "ENERGY_INFRASTRUCTURE"
    PORTS_MARITIME = "PORTS_MARITIME"
    LOGISTICS = "LOGISTICS"
    # Tier 2 — Financial & Economic Systems
    INSURANCE = "INSURANCE"
    FINTECH = "FINTECH"
    CAPITAL_MARKETS = "CAPITAL_MARKETS"
    SOVEREIGN_WEALTH = "SOVEREIGN_WEALTH"
    GOVERNMENT_FINANCE = "GOVERNMENT_FINANCE"
    # Tier 3 — Market & Growth Systems
    ECOMMERCE = "ECOMMERCE"
    CONSTRUCTION = "CONSTRUCTION"
    MANUFACTURING = "MANUFACTURING"
    TOURISM = "TOURISM"
    RETAIL = "RETAIL"
    # Tier 4 — Future & Strategic Systems
    AI_TECHNOLOGY = "AI_TECHNOLOGY"
    STARTUP_ECOSYSTEM = "STARTUP_ECOSYSTEM"
    CYBERSECURITY = "CYBERSECURITY"
    SUSTAINABILITY = "SUSTAINABILITY"


class GDPComponentCode(str, Enum):
    HOUSEHOLD_CONSUMPTION = "HOUSEHOLD_CONSUMPTION"
    BUSINESS_INVESTMENT = "BUSINESS_INVESTMENT"
    GOVERNMENT_SPENDING = "GOVERNMENT_SPENDING"
    NET_EXPORTS = "NET_EXPORTS"


class LanguageVariant(BaseModel):
    """Bilingual content container."""
    en: str
    ar: str = ""


class ScenarioInput(BaseModel):
    """Input scenario that triggers the intelligence pipeline."""
    id: str
    title: LanguageVariant
    description: LanguageVariant
    category: ScenarioCategory
    severity: ScenarioSeverity
    affected_countries: list[GCCCountryCode]
    affected_sectors: list[SectorCode]
    source_count: int = Field(ge=0, default=0)
    confidence: float = Field(ge=0.0, le=1.0)
    created_at: Optional[str] = None


class ScenarioResponse(ScenarioInput):
    """Scenario with computed stress level."""
    stress_level: float = Field(ge=0.0, le=100.0, description="Overall scenario stress 0-100")

File: apps/executive-dashboard/src/types/scenario.ts
/**
 * Scenario Types — exact mirror of schemas/scenario.py
 *
 * Source: services/intelligence-api/app/schemas/scenario.py
 * Contract: every field, every enum, every constraint must match.
 */

// ─── Enums ──────────────────────────────────────────────────────

export type ScenarioSeverity = "critical" | "high" | "medium" | "low";

export type ScenarioCategory =
  | "trade"
  | "energy"
  | "geopolitical"
  | "financial"
  | "regulatory"
  | "demand"
  | "infrastructure"
  | "confidence";

export type GCCCountryCode = "SA" | "AE" | "KW" | "QA" | "BH" | "OM";

export type SectorCode =
  // Tier 1 — Critical Sovereign
  | "AVIATION"
  | "OIL_GAS"
  | "BANKING"
  | "ENERGY_INFRASTRUCTURE"
  | "PORTS_MARITIME"
  | "LOGISTICS"
  // Tier 2 — Financial & Economic
  | "INSURANCE"
  | "FINTECH"
  | "CAPITAL_MARKETS"
  | "SOVEREIGN_WEALTH"
  | "GOVERNMENT_FINANCE"
  // Tier 3 — Market & Growth
  | "ECOMMERCE"
  | "CONSTRUCTION"
  | "MANUFACTURING"
  | "TOURISM"
  | "RETAIL"
  // Tier 4 — Future & Strategic
  | "AI_TECHNOLOGY"
  | "STARTUP_ECOSYSTEM"
  | "CYBERSECURITY"
  | "SUSTAINABILITY";

export type GDPComponentCode =
  | "HOUSEHOLD_CONSUMPTION"
  | "BUSINESS_INVESTMENT"
  | "GOVERNMENT_SPENDING"
  | "NET_EXPORTS";

// ─── Shared ─────────────────────────────────────────────────────

export interface LanguageVariant {
  en: string;
  ar: string;
}

// ─── Models ─────────────────────────────────────────────────────

export interface ScenarioInput {
  id: string;
  title: LanguageVariant;
  description: LanguageVariant;
  category: ScenarioCategory;
  severity: ScenarioSeverity;
  affected_countries: GCCCountryCode[];
  affected_sectors: SectorCode[];
  source_count: number;
  confidence: number; // 0-1
  created_at?: string;
}

export interface ScenarioResponse extends ScenarioInput {
  stress_level: number; // 0-100
}
